src/emoji/data.js

```diff
@@ -52,7 +52,7 @@ export const getFilteredEmojis = (
   // We start by making a map from matching emoji names to a number
   // representing how good a match it is: 0 for a prefix match, 1 for a
   // match anywhere else in the string.
-  const matchingUnicodeEmoji: Array<[string, number]> = [];
+  const allMatchingEmoji: Map<string, number> = new Map();
```

Comment:
The commit message for this commit says:

    emoji [nfc]: Use for loop for matchingImageEmoji.
    
    Use an honest loop to add matching image emoji to a Map instead of using
    .map().filter().

But the actual commit does more than turn that map/filter into a for-loop -- it also changes the data structures, so that we accumulate directly into a Map here rather than build up two arrays and then make a Map.

My ideal version of this branch would have two separate steps there. (As described at https://github.com/zulip/zulip-mobile/pull/5326#issuecomment-1089435130, and as in the previous version of the branch; in fact that previous version further split the latter step into two.) But having them combined as one commit is OK -- it doesn't become that complicated of a diff.

However: it is important to me that the commit message give an accurate picture of the main point of what's happening in the commit. So if you are going to combine these steps, the commit message needs to describe the second step and not only the first one.

Reply:
I will split this commit. Sorry for the oversight.

Reply:
Commit split.

Reply:
Thanks!